microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/NamefellowMetricsTest.java

```diff
@@ -0,0 +1,126 @@
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Arquillian.class)
+public abstract class NamefellowMetricsTest {
+
+    abstract String getPingOneURL();
+
+    abstract String getPingTwoURL();
+
+    abstract String getPingOneTag();
+
+    abstract String getPingTwoTag();
+
+    /**
+     * @tpTestDetails High level scenario to verify two none-reusable metrics of the same name are registered and tagged
+     * properly. Metrics are separate archives - multiple-deployment / subdeployment scenario.
+     * @tpPassCrit Metrics are tagged properly
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
+    @InSequence(1)
+    public void applicationMetricsAreRegisteredAtDeploymentTime() throws IOException {
+
+        String response = Request.Options("http://localhost:9990/metrics")
+                .addHeader("Accept", "application/json").execute().returnContent().asString();
+        JsonObject json = new JsonParser().parse(response).getAsJsonObject();
+
+        assertThat(json.has("application")).isTrue();
+        JsonObject app = json.getAsJsonObject("application");
+        assertThat(app.size()).isEqualTo(1);
+        assertThat(app.has("ping-count")).isTrue();
+        JsonObject counter = app.getAsJsonObject("ping-count");
+
+        assertThat(counter.has("tags")).isTrue();
+        JsonArray tags = counter.getAsJsonArray("tags");
+        assertThat(tags.size()).isEqualTo(2);
+
+        JsonArray tag1 = tags.get(0).getAsJsonArray();
+        JsonArray tag2 = tags.get(1).getAsJsonArray();
+
+        assertThat(tag1.size()).isEqualTo(1);
+        assertThat(tag2.size()).isEqualTo(1);
+
+        boolean dep1Present = tag1.contains(new JsonPrimitive("_app=" + getPingOneTag())) ^ tag2.contains(new JsonPrimitive("_app=" + getPingOneTag()));
+        boolean dep2Present = tag1.contains(new JsonPrimitive("_app=" + getPingTwoTag())) ^ tag2.contains(new JsonPrimitive("_app=" + getPingTwoTag()));
+        assertThat(dep1Present && dep2Present).isTrue();
+    }
+
+    /**
+     * @tpTestDetails Smoke scenario. Calling servlets which call specific CDI bean. This is needed for furhter testcases.
+     * @tpPassCrit HTTP response contains correct string.
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
+    @InSequence(2)
+    public void trigger() throws IOException {
+        String response = Request.Get(getPingOneURL()).execute().returnContent().asString();
+        assertThat(response).isEqualTo("pong one");
+
+        response = Request.Get(getPingTwoURL()).execute().returnContent().asString();
+        assertThat(response).isEqualTo("pong two");
+
+
+        Request.Get(getPingTwoURL()).execute().discardContent();
+        Request.Get(getPingTwoURL()).execute().discardContent();
+        Request.Get(getPingTwoURL()).execute().discardContent();
+
+        Request.Get(getPingOneURL()).execute().discardContent();
+    }
+
```

Comment:
`@InSequence(3)` missing

and why is it needed to have such strict order ?

Reply:
Fixed.